doc/api/n-api.md

```diff
@@ -251,6 +251,73 @@ NAPI_MODULE_INIT() {
 }
 ```
 
+## Environment Life Cycle APIs
+
+> Stability: 1 - Experimental
+
+[Section 8.7][] of the [ECMAScript Language Specification][] defines the concept
+of an "Agent" as a self-contained environment in which JavaScript code runs.
+Multiple such Agents may be started and terminated either concurrently or in
+sequence by the process.
+
+From the perspective of a native addon this means that the bindings it provides
+may be called multiple times, from multiple contexts, and even concurrently from
+multiple threads.
+
+Native addons may need to allocate global state of which they make use during
+their entire life cycle such that the state must be unique to each instance of
+the addon.
+
+To this env, N-API provides a way to allocate data such that its life cycle is
+tied to the life cycle of the Agent.
```

Comment:
I think it would be useful to explain how the "Agent" concept maps to Node.  I think these are currently an instance of Node (whether started through the executable or "embedded" in another process) and each worker thread started, right?

Reply:
AIUI that is correct.